util/src/main/java/io/kubernetes/client/util/Config.java

```diff
@@ -114,16 +114,22 @@ public static ApiClient fromConfig(Reader input) {
             ex.printStackTrace();
         }
 
-        // It's silly to have to do it in this order, but each SSL setup
-        // consumes the CA cert, so if we do this before the client certs
-        // are injected the cert input stream is exhausted and things get
-        // grumpy'
-        String caCert = config.getCertificateAuthorityData();
-        String caCertFile = config.getCertificateAuthorityFile();
-        try {
-            client.setSslCaCert(SSLUtils.getInputStreamFromDataOrFile(caCert, caCertFile));
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
+        if (config.verifySSL()) {
+            // It's silly to have to do it in this order, but each SSL setup
+            // consumes the CA cert, so if we do this before the client certs
+            // are injected the cert input stream is exhausted and things get
+            // grumpy'
+            String caCert = config.getCertificateAuthorityData();
+            String caCertFile = config.getCertificateAuthorityFile();
+            if (caCert != null || caCertFile != null) {
+                try {
+                    client.setSslCaCert(SSLUtils.getInputStreamFromDataOrFile(caCert, caCertFile));
+                } catch (FileNotFoundException e) {
+                    e.printStackTrace();
+                }
+            }
+        } else {
+            client.setVerifyingSsl(false);
```

Comment:
This call is static and always creates both a new KubeConfig object and returns a new client, so I don't think that is a risk.